mlos_core/mlos_core/data_classes.py

```diff
@@ -0,0 +1,365 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""Data classes for mlos_core used to pass around configurations, observations, and
+suggestions.
+"""
+from __future__ import annotations
```

Comment:
Thing I noted earlier that we shouldn't need this if we use `"QuotedClassNameAsTypeHint"`

Reply:
The linter gets pretty unhappy about this since you are using the constructors ect

Reply:
Fixed